src/Lookup/MultiplexingEntityLookup.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\EntityDocument;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikimedia\Assert\Assert;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterElementTypeException;
+
+/**
+ * Wrapper around repository-specific EntityLookups
+ * picking up the right lookup object for the particular input.
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class MultiplexingEntityLookup implements EntityLookup {
+
+	/**
+	 * @var EntityLookup[]
+	 */
+	private $lookups;
+
+	/**
+	 * @param EntityLookup[] $lookups associative array with repository names (strings) as keys
+	 *                                and EntityLookup objects as values. Empty-string key
+	 *                                defines lookup for the local repository.
+	 *
+	 * @throws ParameterAssertionException
+	 * @throws ParameterElementTypeException
```

Comment:
It's not so nice to force clients to deal with both as they are caused by very similar scenarios. Do they have a common supertype that can be used here instead?